README.md

```diff
@@ -1 +1,131 @@
+- `omnibus_accounts:` `AccountId[]` - list of all chain omnibus accounts, e.g. `omni@chain_a`, `omni@chain_b`
+- `domestic_torii_url: Url`, `hub_torii_url: Url` - URLs of the two chains
+- `domestic_omnibus_account: AccountId` - account associated with the connected domestic chain (needed as a target on the hub chain)
+
+### Relay function
+
+- Subscribe to domestic chain transfers targeting _all_ omnibus accounts[^1]
+  - When `Transfer(AssetId, TargetOmnibusAccountId, Numeric)` with metadata `destination: AccountId` where `AssetId` = `AssetDefinitionId` + source `AccountId` happens:
+    - On Hub chain, create `Transfer(AssetId, TargetOmnibusAccountId, Numeric)` with metadata `destination: AccountId` (forwarded) where `AssetId` = `AssetDefinitionId` + _domestic omnibus_ `AccountId`
+- Subscribe to hub chain transfers targeting _only_ domestic omnibus account.
+  - Extract source omnibus `AccountId` and destination `AccountId`, and make a `Transfer(SourceOmnibusAssetId, DestinationAccountId, Numeric)`
+
+### Limitations
+
+This is an intentionally simple design.
+
+- Relays must be trusted
+- Since relays are subscribing to events stream, it is possible to skip a transfer event in case of a race condition/connection issues
+
+### Demo
+
+Configuration:
+
+- 3 domestic chains + 1 hub chain.
+- 3 user accounts on each domestic chain (9 accounts total).
+- **Same** 3 assets on all chains. E.g. each domestic chain would have an asset definition `rose#wonderland`.
+- Assign some starting balances for accounts
+
+UI:
+
+- Display balances of each account on each chain, update in real time
+- Display block height of each chain, update in real time
+- Provide a form to initiate a **cross-chain** transfer: from any account, to any account, for any asset id and quantity
+
+## Stage 2: Introduce prover-verifier
+
+This would significantly increase complexity of the design.
+
+Core ideas:
+
+- Relays are not trusted
+- Deploy a **verifier** as a trigger (WASM) on each chain. It is a decentralised on-chain logic.
+- Verifier has knowledge of other chain's validators set (set of public keys)
+- Relay must submit block headers subsequently to the verifier trigger (both from domestic chain to hub chain and vice versa). Verifier ensures the block header is a valid continuation (height match, hash match, signed by known validators)
+- Relay must submit _proofs_ of transfer intentions on the source chain to the verifier on the target chain
+- Proof is, roughly, a transaction payload which contains the instruction (or multiple ones) and a **merkle proof** of this transaction being a part of the latest block. Payload is sufficient to get transaction's _hash_. Transaction must be in the latest block (whose header was submitted to the verified previously). Since block header contains merkle tree of the transactions, verifier can validate that the given transaction is indeed a part of the given block, which is valid.
```

Comment:
The block header contains only the Merkle root.
In this way, the verifier performs verification based on less and limited information compared to the prover, which is why it is called a "light" client.